apiserver/facades/client/spaces/move.go

```diff
@@ -0,0 +1,289 @@
+		ops = append(ops, subnet.UpdateSpaceOps(o.space.Id())...)
+	}
+	return ops, nil
+}
+
+// Done (ModelOperation) is called upon execution of the operations returned by
+// Build. It records the successfully moved subnets for later retrieval.
+func (o *moveSubnetsOp) Done(err error) error {
+	if err != nil {
+		o.results = make([]MovedSubnet, len(o.subnets))
+		for i, subnet := range o.subnets {
+			mc := MovedSubnet{
+				ID:        subnet.ID(),
+				FromSpace: subnet.SpaceName(),
+			}
+			o.results[i] = mc
+		}
+	}
+	return err
+}
+
+// GetMovedSubnets (MoveSubnetsOp) returns the results of successfully
+// executed movement of subnets to a new space.
+func (o *moveSubnetsOp) GetMovedSubnets() []MovedSubnet {
+	return o.results
+}
+
+// MoveSubnets ensures that the input subnets are in the input space.
+func (api *API) MoveSubnets(args params.MoveSubnetsParams) (params.MoveSubnetsResults, error) {
+	var result params.MoveSubnetsResults
+
+	if err := api.ensureSpacesAreMutable(); err != nil {
+		return result, err
+	}
+
+	results := make([]params.MoveSubnetsResult, len(args.Args))
+	for i, toSpaceParams := range args.Args {
+		// Note that although spaces have an ID, a space tag represents
+		// a space *name*, which remains a unique identifier.
+		// We need to retrieve the space in order to use its ID.
+		spaceTag, err := names.ParseSpaceTag(toSpaceParams.SpaceTag)
+		if err != nil {
+			results[i].Error = common.ServerError(errors.Trace(err))
+			continue
+		}
+		spaceName := spaceTag.Id()
+
+		subnets, err := api.getMovingSubnets(toSpaceParams.SubnetTags)
+		if err != nil {
+			result.Results[i].Error = common.ServerError(errors.Trace(err))
+			continue
+		}
+
+		if err := api.ensureSubnetCanBeMoved(subnets, spaceName, toSpaceParams.Force); err != nil {
+			results[i].Error = common.ServerError(errors.Trace(err))
+			continue
+		}
+
+		operation, err := api.opFactory.NewMoveSubnetsOp(spaceName, subnets)
+		if err != nil {
+			results[i].Error = common.ServerError(errors.Trace(err))
+			continue
+		}
+
+		if err = api.backing.ApplyOperation(operation); err != nil {
+			results[i].Error = common.ServerError(errors.Trace(err))
+			continue
+		}
+
+		results[i].NewSpaceTag = spaceTag.String()
+		results[i].MovedSubnets = paramsFromMovedSubnet(operation.GetMovedSubnets())
+	}
+
+	result.Results = results
+	return result, nil
+}
+
+// getMovingSubnets acquires all the subnets that we have
+// been requested to relocate, identified by their tags.
+func (api *API) getMovingSubnets(tags []string) ([]MovingSubnet, error) {
+	subnets := make([]MovingSubnet, len(tags))
+	for i, tag := range tags {
+		subTag, err := names.ParseSubnetTag(tag)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		subnet, err := api.backing.MovingSubnet(subTag.Id())
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		subnets[i] = subnet
+	}
+	return subnets, nil
+}
+
+// TODO (manadart 2020-03-19): At this point we only restrict and notify
```

Comment:
This part is quite important and will follow. I anticipate that we will simply be able to add another call to this method that checks the endpoint bindings.